cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -128,6 +125,22 @@ func (p *BicepProvider) EnsureEnv(ctx context.Context) error {
 		return fmt.Errorf("compiling bicep template: %w", err)
 	}
 
+	locationFilterFn := func(_ account.Location) bool { return true }
+	if locationParam, defined := template.Parameters["location"]; defined {
```

Comment:
When I was prototyping this, I tried to figure out what the name of the location parameter was by looking at the `.parameters.json` file to see what parameter `AZURE_LOCATION` was used to influence.

With the addition of bicepparam now, that logic is harder to replicate, but I think assuming that the `location` parameter, if present, controls the location that is used is a reasonable huristic.  Much simpler too!